cdap-common/src/main/java/co/cask/cdap/common/service/AbstractRetryableScheduledService.java

```diff
@@ -153,7 +163,9 @@ protected final void runOneIteration() throws Exception {
         e.addSuppressed(new RetriesExhaustedException(String.format("Retries exhausted after %d failures and %d ms.",
                                                                     failureCount,
                                                                     System.currentTimeMillis() - nonFailureStartTime)));
-        throw e;
+        delayMillis = handleRetriesExhausted(e);
```

Comment:
Should take a `Maths.max(0L, handleRetriesExhausted(e))` to guard against -ve number.